### PR TITLE
DB/Script - Add bat mini event to hourly bell event in Brill

### PR DESCRIPTION
**Pull request by schell244**
## Changes Proposed:
Ported Brill bat event from VMangos vmangos/core@373845a

I've added some small cosmetic improvement to the script: 
- Bats start flying from clock tower AFTER the first bell toll has happend.

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6126

## SOURCE:
Credits to VMangos

## Tests Performed:
run sql, build, tested ingame

## How to Test the Changes:
1. go to Brill
2. wait for full hour XX:00 (or start event manually .event start 73)
3.  Bats spawn and fly in the direction of the zeppelin tower, play the sound again mid-path, and land on the roof of that small building at the base of the tower. After most of them land, they play the sound yet again and fly away/despawn.

## Known Issues and TODO List:
- VMangos has set some additional creature_template flags, which cause errors with azerothcore. Those flags don't affect the script from working.
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1636977376771149000.sql [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+(@PATH_BAT_GUID_1, 8,  2163.419922, 392.268005, 60.324600, 100.000000, 0, 2, 0, 0, 0),                     -- spline 6/11
+(@PATH_BAT_GUID_1, 9,  2119.560059, 369.352997, 60.324600, 100.000000, 0, 2, 0, 0, 0),                     -- spline 7/11
+(@PATH_BAT_GUID_1, 10,  2096.479980, 331.221985, 69.324600, 100.000000, 0, 2, 0, 0, 0),                    -- spline 8/11
+(@PATH_BAT_GUID_1, 11, 2081.800049, 305.434998, 80.074600, 100.000000, 0, 2, 0, 0, 0),                     -- spline 9/11
+(@PATH_BAT_GUID_1, 12, 2062.149902, 278.320007, 75.241302, 100.000000, 0, 2, 0, 0, 0),                     -- spline 10/11
+(@PATH_BAT_GUID_1, 13, 2058.270020, 272.925995, 68.741302, 100.000000, 60000, 2, 0, 0, 0);                 -- spline 11/11
+
+-- Add sound script
+DELETE FROM `waypoint_scripts` WHERE `id`= @BAT_SOUND_ACTION_1;
+INSERT INTO `waypoint_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `dataint`, `x`, `y`, `z`, `o`, `guid`) VALUES(@BAT_SOUND_ACTION_1, 0, 16, 6596 , 2 , 0 , 0 , 0 , 0 , 0 , @BAT_GUID_1);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1636977376771149000.sql:104]: Maybe a line break after VALUE for improve readability?

**Review by Nyeriah — APPROVED** [review]

File: data/sql/updates/pending_db_world/rev_1636977376771149000.sql [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+(@PATH_BAT_GUID_7, 9,  2212.919922, 296.894989, 45.647999, 100.000000, 0, 2, 0, 0, 0),                     -- spline 7/8
+(@PATH_BAT_GUID_7, 10,  2181.479980, 318.591003, 46.897999, 100.000000, 0, 2, 0, 0, 0),                    -- spline 8/8
+(@PATH_BAT_GUID_7, 11, 2144.870117, 327.882996, 71.922401, 100.000000, 0, 2, @BAT_SOUND_ACTION_2, 100, 0), -- spline 0/3
+(@PATH_BAT_GUID_7, 12, 2110.179932, 313.437988, 88.172501, 100.000000, 0, 2, 0, 0, 0),                     -- spline 1/3
+(@PATH_BAT_GUID_7, 13, 2101.520020, 321.703003, 79.090698, 100.000000, 0, 2, 0, 0, 0),                     -- spline 2/3
+(@PATH_BAT_GUID_7, 14, 2071.030029, 289.042999, 70.729599, 100.000000, 60000, 2, 0, 0, 0);                 -- spline 3/3
+
+-- Add sound script
+DELETE FROM `waypoint_scripts` WHERE `id`= @BAT_SOUND_ACTION_2;
+INSERT INTO `waypoint_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `dataint`, `x`, `y`, `z`, `o`, `guid`) VALUES(@BAT_SOUND_ACTION_2, 0, 16, 6596 , 2 , 0 , 0 , 0 , 0 , 0 , @BAT_GUID_7);
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1636977376771149000.sql:211]: on more line break for read ability

**Comment by acidmanifesto** [comment]
besides one more line break just for readability sake. this sql does compliment the event of the hourly bells very well and there is no issues that i can see.

**Comment by acidmanifesto** [comment]
@schell244 i really hate to do this to you. but with my recent pr, i changed the isLarge collumn to visibilityDistanceType with the creature_addon at the end of your sql.
My PR that got merge that will changes it for reference. 
https://github.com/azerothcore/azerothcore-wotlk/pull/9180/files

**Comment by schell244** [comment]
@acidmanifesto no problem, I've just updated the creature_addon lines to use `visibilityDistanceType`. I've set the value to 3 (Large) and did a quick ingame check. Works fine for me
Unbenannt
.

**Comment by betayi** [comment]
In WorldBase SQL, the guid 201193/201194 was used for NPC in ICC map, how about these NPC ?

**Comment by Nyeriah** [comment]
@schell244 
```sql
SELECT * FROM creature WHERE guid IN(201203, 201194);
```

Those guids were in use already. Could you please restore those creatures?

**Comment by schell244** [comment]
I'll restore them this evening. When I created this PR those GUID's appeared to be free in my db, but that was obviously wrong:eyes:

**Comment by acidmanifesto** [comment]
@Nyeriah 
```
DELETE FROM `creature` WHERE `guid`=201194;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (201194, 38054, 631, 0, 0, 15, 1, 0, 0, -46.5868, 2251.06, 30.7375, 3.83972, 86400, 0, 0, 17880, 8814, 0, 0, 0, 0, '', 0);
DELETE FROM `creature` WHERE `guid`=201203;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (201203, 2110, 631, 0, 0, 15, 1, 0, 0, 16.1163, 2211.13, 30.199, 0.679835, 86400, 0, 0, 1, 0, 0, 0, 0, 0, '', 0);
```
@schell244  sorry about the collumn change as well. i been attempting to move the core forward.

**Comment by acidmanifesto** [comment]
also in regards to those two guids, i found **NO conflict** in waypoint_data, waypoint_scripts, or creature_addon.

**Comment by Nyeriah** [comment]
> @Nyeriah here are the missing guids. i guess we need to realight some bats to a differ id entirely.
> 
> ```
> DELETE FROM `creature` WHERE `guid`=201194;
> INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (201194, 38054, 631, 0, 0, 15, 1, 0, 0, -46.5868, 2251.06, 30.7375, 3.83972, 86400, 0, 0, 17880, 8814, 0, 0, 0, 0, '', 0);
> DELETE FROM `creature` WHERE `guid`=201203;
> INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (201203, 2110, 631, 0, 0, 15, 1, 0, 0, 16.1163, 2211.13, 30.199, 0.679835, 86400, 0, 0, 1, 0, 0, 0, 0, 0, '', 0);
> ```
> 
> @schell244 sorry about the collumn change as well. i been attempting to move the core forward.

Hey, thanks for trying to help, but those guids are now used by the bats, so we gotta find two unused guids for the ICC vendors.

**Comment by schell244** [comment]
Thanks @acidmanifesto for providing the missing creature spawns! I just did a fresh world db install (without my spawn patch) and both guids (201194 and 201203) are unused in my db (I've used the auto setup method for getting a fresh world db). Are those vendors from a custom sql patch? When I change their guids to something like 201206 and 201207, how can I verify that those guids are actually free?

**Comment by Nyeriah** [comment]
> I just did a fresh world db install (without my spawn patch) and both guids (201194 and 201203) are unused in my db (I've used the auto setup method for getting a fresh world db). Are those vendors from a custom sql patch? When I change their guids to something like 201206 and 201207, how can I verify that those guids are actually free?

That's impossible

1) the ICC vendors were already there in my database
2) those guids are now used by the very own npcs you added in your script

Your database is missing updates, startup worldserver and try again (if you got autoupdater on)

**Comment by schell244** [comment]
Turns out my sql tool (SQlyog Community) did not show those guids when using the UI button filter. I found the spawn now by executing a search query.

**Comment by schell244** [comment]
Sorry for the mess. Fix is here: https://github.com/azerothcore/azerothcore-wotlk/pull/9250 Thanks all for your help!

**Pull request merged**